src/main/scala/outwatch/dom/VDomModifier.scala

```diff
@@ -32,9 +32,25 @@ final case class Style(title: String, value: String) extends Attribute
 final case class Key(value: String) extends Property
 
 sealed trait Hook extends Property
-final case class InsertHook(observer: Observer[Element]) extends Hook
-final case class DestroyHook(observer: Observer[Element]) extends Hook
-final case class UpdateHook(observer: Observer[(Element, Element)]) extends Hook
+
+sealed trait HookSingle extends Hook {
```

Comment:
Can we make these traits generic? So we have `def observer: Observer[T]`. We can later pattern match on the value when mapping.

Reply:
If there is a way to make this trait generic and combine the functions `createHookSingle` etc. , great. But if that turns out to be impractical, I'm fine with this solution.